### PR TITLE
Remove empty catch blocks and remove allow-empty-catch tslint rule

### PR DESCRIPTION
**Pull request by djsmith85**
## Type of change
- [ ] Bug fix
- [ ] New feature development
- [X] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective
Removes the allow-empty-catch exception in tslint and adds logging to most of the catch statements. This is to avoid quietly swallowing errors which makes it difficult to debug errors.

## Code changes
* **jslib:** Updated submodule to https://github.com/bitwarden/jslib/commit/f09fb69882525b3be7b2e257e7723eeb79b343d1
* **app/app.components.ts:** Pass logService into constructor and log error on catch
* **app/accounts/apiKey.components.ts:** Pass logService into constructor and log error on catch
* **app/tabs/settings.components.ts:** Pass logService into constructor and log error on catch
* **app/services/services.module.ts:** Fix build error after updating of jslib: Created missing policyService
* **services/ldap-directory.service.ts:** Log error on catch
* **src/main.ts:** Log error on catch
* **src/bwdc.ts:** Fix build error after updating of jslib: Created missing policyService
* **program.ts:** Fix build error after updating of jslib: Add missing params to LoginCommand ctor
* **tslint.json:** Removed rule to allow empty catch statements

## Before you submit
- [ ] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -1,17 +1,39 @@
 {
   "extends": "tslint:recommended",
   "rules": {
-    "align": [ true, "statements", "members" ],
+    "align": [
```
Comment (Hinton) [inline tslint.json:4]: Not sure if we want to format this file, it's mostly identical across our projects

File: src/bwdc.ts [hunk 2]
```diff
@@ -108,6 +127,25 @@ export class Main {
...
+        this.collectionService = new CollectionService(this.cryptoService, this.userService, this.storageService,
+            this.i18nService);
+        this.sendService = new SendService(this.cryptoService, this.userService, this.apiService, this.fileUploadService, this.storageService,
+            this.i18nService, this.cryptoFunctionService);
+
+        this.loginSyncService = new LoginSyncService(this.userService, this.apiService, this.settingsService,
+            this.folderService, this.cipherService, this.cryptoService, this.collectionService, this.storageService,
+            this.messagingService, this.policyService, this.sendService, this.logService,
+            async (expired: boolean) => this.messagingService.send('logout', { expired: expired }));
+
```
Comment (Hinton) [inline src/bwdc.ts:148]: Do we need all these services for the directory connector?

Comment (djsmith85) [reply src/bwdc.ts:148]: Unfortunately yes. These svc-instances are needed to meet all the dependencies of for example the LoginCommand, which needs the LoginSyncService. Had to include them to get rid of all the build errors.

File: tslint.json (hunk 1 above)
Comment (djsmith85) [reply tslint.json]: Reverted the formatting

**Review by Hinton — APPROVED** [review]

**Pull request merged**